### PR TITLE
Make tests more resilient to crashing failures.

### PR DESCRIPTION
**Pull request by jverkoey**

### CODE REVIEW

File: tests/unit/BeginFromCurrentStateTests.swift [hunk 1]
```diff
@@ -240,10 +240,12 @@ class BeginFromCurrentStateTests: XCTestCase {
...
-    XCTAssertEqual(animation.keyPath, AnimatableKeyPath.opacity.rawValue)
-    XCTAssertEqual(animation.fromValue as! Float, initialValue)
-    XCTAssertEqualWithAccuracy(animation.toValue as! CGFloat, 1.0, accuracy: 0.0001)
+    if addedAnimations.count == 2 {
+      let animation = addedAnimations.last as! CABasicAnimation
+      XCTAssertFalse(animation.isAdditive)
+      XCTAssertEqual(animation.keyPath, AnimatableKeyPath.opacity.rawValue)
+      XCTAssertEqual(animation.fromValue as! Float, initialValue)
+      XCTAssertEqualWithAccuracy(animation.toValue as! CGFloat, 1.0, accuracy: 0.0001)
+    }
```
Comment (romoore) [inline tests/unit/BeginFromCurrentStateTests.swift:249]: Should this be paired with an else block with an `XCTAssertTrue(false)`?

File: tests/unit/ImplicitAnimationTests.swift [hunk 2]
```diff
@@ -78,6 +78,9 @@ class ImplicitAnimationTests: XCTestCase {
     }
 
     XCTAssertEqual(addedAnimations.count, 1)
+    guard addedAnimations.count == 1 else {
+      return
```
Comment (romoore) [inline tests/unit/ImplicitAnimationTests.swift:82]: Should this include an `XCTAssertTrue(false)` to mark the test as failed?  Same question for the `guard`/`if-let` statements below.

File: tests/unit/BeginFromCurrentStateTests.swift (hunk 1 above)
Comment (jverkoey) [reply tests/unit/BeginFromCurrentStateTests.swift:249]: This is paired with the XCTAssertEqual above. If that assertion fails, this logic won't be executed.

File: tests/unit/ImplicitAnimationTests.swift (hunk 2 above)
Comment (jverkoey) [reply tests/unit/ImplicitAnimationTests.swift:82]: Similar to above, this is an early exit on the condition that our XCTAssertEqual assertion fails.

Comment (romoore) [reply tests/unit/ImplicitAnimationTests.swift:82]: Ah, I see them. Sorry&mdash;still finishing my coffee this morning...

**Review by romoore — APPROVED** [review]
LGTM!

**Pull request merged**